internal/blocksync/pool.go

```diff
@@ -78,9 +79,10 @@ type BlockPool struct {
 	requesters map[int64]*bpRequester
 	height     int64 // the lowest key in requesters.
 	// peers
-	peers         map[p2p.ID]*bpPeer
-	sortedPeers   []*bpPeer // sorted by curRate, highest first
-	maxPeerHeight int64     // the biggest reported height
+	peers       map[p2p.ID]*bpPeer
+	sortedPeers []*bpPeer // sorted by curRate, highest first
+
+	maxPeerHeight atomic.Int64 // the biggest reported height
```

Comment:
Why do we need to turn this into an atomic? Don't we have a lock already? I mean, is there evidence we're improving contention?

Reply:
No hard evidence. https://github.com/cometbft/cometbft/pull/3134#discussion_r1617918338

Do you think I should revert?